apluslms_roman/builder.py

```diff
@@ -10,18 +10,24 @@
 
 class Builder:
     def __init__(self, engine, config, observer=None):
-        assert isdir(config.__path__), "Course configuration path doesn't exists"
+        if not isdir(config.dir):
+            raise ValueError("config.dir isn't a directory.")
         self.config = config
-        self.path = config.__path__
+        self.path = config.dir
         self._engine = engine
         self._observer = observer or StreamObserver()
 
-    def build(self):
+    def buildSteps(self):
+        return [BuildStep.from_config(i, step) for i, step in enumerate(self.config.steps)]
+
+    def build(self, step_list=[]):
         backend = self._engine.backend
         observer = self._observer
-        steps = [BuildStep.from_config(step) for step in self.config.steps]
+        steps = self.buildSteps()
+        if (step_list):
```

Comment:
no brackets needed in python :)

Reply:
this is why I shouldn't code with other languages... I keep trying to call functions without brackets because of scala